src/main/java/rookiesspring/controller/admin/CategoryAdminController.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
+ * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
+ */
+package rookiesspring.controller.admin;
+
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import rookiesspring.dto.CategoryDTO;
+import rookiesspring.dto.update.CategoryUpdateDTO;
+import rookiesspring.service.CategoryService;
+import rookiesspring.util.Util;
+
+/**
+ *
+ * @author HP
+ * @author Tamina
+ */
+@RestController
+@RequestMapping("/api/v1/admin/categories")
+public class CategoryAdminController {
+
+    CategoryService service;
+
+    public CategoryAdminController(CategoryService service) {
+        this.service = service;
+    }
+
+    @GetMapping({"", "/"})
+    public ResponseEntity getAllCategory(@RequestParam(defaultValue = "", required = false) String name) {
+        return ResponseEntity.ok(service.findAll(name));
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity getOneCategory(@PathVariable(name = "id") long id) {
+        return ResponseEntity.ok(service.findById(id));
+    }
+
+    @GetMapping("/products")
+    public ResponseEntity getAllCategoriesWithProducts(@RequestParam(required = false) String name) {
+        return ResponseEntity.ok(service.findAllFull(name));
```

Comment:
You need to pay attention to naming functions so that they are clear and meaningful, eg: findAllFull, findByIdFull,…